j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/BytecodeTask.java

```diff
@@ -67,14 +77,37 @@ public Task resolve(Project project, Config config) {
         // be if we had built a jar
         Input resources = input(project, OutputTypes.INPUT_SOURCES).filter(NOT_BYTECODE);
 
-        List<Input> bytecodeClasspath = scope(project.getDependencies(), com.vertispan.j2cl.build.task.Dependency.Scope.COMPILE)
+        List<Input> bytecodeClasspath = scope(project.getDependencies().stream().filter(dependency -> !dependency.isAPT()).collect(Collectors.toSet()),
+                com.vertispan.j2cl.build.task.Dependency.Scope.COMPILE)
+                .stream()
+                .map(inputs(OutputTypes.BYTECODE))
+                .collect(Collectors.toUnmodifiableList());
+
+        List<Input> inReactorProcessors = scope(project.getDependencies().stream().filter(dependency -> dependency.getProject().hasSourcesMapped()
+                        && !dependency.getProject().isJsZip()).collect(Collectors.toSet()),
+                com.vertispan.j2cl.build.task.Dependency.Scope.COMPILE)
                 .stream()
                 .map(inputs(OutputTypes.BYTECODE))
                 .collect(Collectors.toUnmodifiableList());
```

Comment:
There should have the filter also done here, so you can spot the exact processor files up front - technically maybeAddInReactorAptProcessor shouldnt even be able to call filter(), but i failed to guard that correctly.
```suggestion
                .map(inputs(OutputTypes.BYTECODE))
                .map(input -> input.filter(APT_PROCESSOR))
```
This will limit `inReactorProcessors` from containing anything except `META-INF/services/javax.annotation.processing.Processor`, which should also let caching be a bit more clever about what has changed (in some cases, but probably not this one).

Reply:
At this point, we can't be sure if the reactor annotation processor is compiled; so there is no META-INF/services/javax.annotation.processing.Processor created. So the only option is to check it later when the deps processing is finished.

Reply:
That's okay, right? The resulting filtered input will just be empty if there are no processors - if non-empty, you've got a file to read to see which processors are present.

Reply:
ok, let me check, but I think in this case this List will be empty, even there are reactor processors

Reply:
Empty is fine - you will see that nothing is in that filesAndHashes, and so don't have to read anything, right?

If that doesn't work, then most of these tasks (esp incremental ones) won't work at all with new files...